src/main/java/com/ldtteam/structurize/placement/StructurePlacer.java

```diff
@@ -399,9 +399,22 @@ public BlockPlacementResult getResourceRequirements(
     {
         final BlockState worldState = world.getBlockState(worldPos);
         boolean sameBlockInWorld = false;
-        if (worldState.getBlock() == localState.getBlock() && tileEntityData == null)
+        if (worldState.getBlock() == localState.getBlock())
         {
-            sameBlockInWorld = true;
+            if (tileEntityData == null)
+            {
+                sameBlockInWorld = true;
+            }
+            else
+            {
+                sameBlockInWorld = ItemStackUtils.compareItemStacksIgnoreStackSize(
```

Comment:
We would need the same logic in the areBlocksEqual handling. How do we currently compare DO blocks there atm? This will just skip DO blocks, the real fix should be there.